src/compiler/checker.ts

```diff
@@ -32023,14 +32045,46 @@ namespace ts {
         }
 
         function checkInExpression(left: Expression, right: Expression, leftType: Type, rightType: Type): Type {
-            if (leftType === silentNeverType || rightType === silentNeverType) {
-                return silentNeverType;
+            if (isPrivateIdentifier(left)) {
+                const lexicallyScopedSymbol = lookupSymbolForPrivateIdentifierDeclaration(left.escapedText, left);
+                if (lexicallyScopedSymbol === undefined) {
+                    if (!getContainingClass(left)) {
+                        error(left, Diagnostics.Private_identifiers_are_not_allowed_outside_class_bodies);
+                    }
+                    else {
+                        const suggestion = getSuggestedSymbolForNonexistentProperty(left, rightType);
+                        if (suggestion) {
+                            const suggestedName = symbolName(suggestion);
+                            error(left, Diagnostics.Cannot_find_name_0_Did_you_mean_1, diagnosticName(left), suggestedName);
+                        }
+                        else {
+                            error(left, Diagnostics.Cannot_find_name_0, diagnosticName(left));
+                        }
+                    }
+                    return anyType;
+                }
+
+                markPropertyAsReferenced(lexicallyScopedSymbol, /* nodeForCheckWriteOnly: */ undefined, /* isThisAccess: */ false);
+                getNodeLinks(left.parent).resolvedSymbol = lexicallyScopedSymbol;
+                if (rightType === silentNeverType) {
+                    return silentNeverType;
+                }
+            }
+            else {
+                if (leftType === silentNeverType || rightType === silentNeverType) {
+                    return silentNeverType;
+                }
+                leftType = checkNonNullType(leftType, left);
+                // TypeScript 1.0 spec (April 2014): 4.15.5
+                // Require the left operand to be of type Any, the String primite type, or the Number primite type.

```

Comment:
Missed this bit: 
```suggestion
                // Require the left operand to be of type Any, the String primitive type, or the Number primitive type.
```

Reply:
Fixed. Serves me right for manually doing the suggestions instead of doing it via GH.